observability-test/spanner.ts

```diff
@@ -815,8 +966,161 @@ describe('ObservabilityOptions injection and propagation', async () => {
         expectedEventNames,
         `Unexpected events:\n\tGot:  ${actualEventNames}\n\tWant: ${expectedEventNames}`
       );
+    });
+  });
+});
 
-      done();
+describe('Regression tests for fixed bugs', () => {
```

Comment:
use a better name to describe the test block. Maybe something like `E2E traces with async/await` ?

Reply:
But they are really regression tests, wouldn't E2E be even more obscure? Regression to catch any future bugs given an exact use case per bug.

Reply:
Also btw within this section I've included the callback variant of that bug reproducer